eulerpi/examples/heat/heat.py

```diff
@@ -0,0 +1,210 @@
+        """Getter for the parameter limits.
+
+        Returns:
+            np.ndarray: An array containing the lower and upper limits for each parameter.
+        """
+        return self.param_limits
+
+    @classmethod
+    def forward(self, param: np.ndarray) -> np.ndarray:
+        """Forward method for the heat model. Yields the solution of the anisotropic heat conduction equation at time :math:`\\t=0.1`
+        in five spacial points, which are arranged similar to the number "five" on a dice.
+
+        Args:
+            param (np.ndarray): Entries of the conductivity matrix: param[0] = :math:`\\kappa_{11}`, param[1] = :math:`\\kappa_{22}`, param[2] = :math:`\\kappa_{12}`
+
+        Returns:
+            np.ndarray: The solution of the anisotropic heat conduction equation at time :math:`\\t=0.1`
+        in five spacial points, which are arranged similar to the number "five" on a dice.
+        """
+
+        solution = self.perform_simulation(self, param)
+        # return the solution at four evaluation points
+        eval_points = jnp.multiply(
+            jnp.array(
+                [
+                    [0.25, 0.25, 0.25],
+                    [0.25, 0.75, 0.25],
+                    [0.25, 0.25, 0.75],
+                    [0.25, 0.75, 0.75],
+                ]
+            ),
+            jnp.array(
+                [solution.shape[0], solution.shape[1], solution.shape[2]]
+            ),
+        ).astype(int)
+        sim_res = jnp.array(
+            solution[eval_points[:, 0], eval_points[:, 1], eval_points[:, 2]]
+        )
+        return sim_res
+
+    def param_is_within_domain(self, param: np.ndarray) -> bool:
+        """Checks whether a parameter is within the parameter domain of the model.
+        This condition stems from thermodynamical considerations.
+
+        Args:
+            param(np.ndarray): The parameter to check.
+
+        Returns:
+            bool: True if the parameter is within the limits.
+
+        """
+        return param[0] * param[1] > param[2] ** 2
+
+    def perform_simulation(self, param: np.ndarray) -> np.ndarray:
+        """Performs a simulation of the heat equation with the given parameters.
+
+        Args:
+            param (np.ndarray): Entries of the conductivity matrix: param[0] = :math:`\\kappa_{11}`, param[1] = :math:`\\kappa_{22}`, param[2] = :math:`\\kappa_{12}`
+
+        Returns:
+            np.ndarray: An array containing the solution of the anisotropic heat conduction equation, where the first two indices correspond to the x and y coordinates, respectively, and the third index corresponds to the time.
+        """
+        # set up physical properties
+        time_span = np.array([0, 0.1])
+        plate_length = np.array([1, 1])
+
+        # define the grid
+        num_grid_points = 20
+        dx = plate_length[0] / num_grid_points
+        dy = plate_length[1] / num_grid_points
+        safety_factor = 1.25
+        dt = min(dx, dy) ** 2 / (
```

Comment:
Can you explain this line of code with a comment? E.g.
Stability condition for the heat equation in two dimensions: $\Delta t \leq \frac{h_{min}^2}{(1 + z) * 4 * K}$
I am not sure, what exactly you have to insert for `K`, why the second column of kappa? Use Greschgorin?

Reply:
Will do! The second column of the param limits are the highest possible values which are the most limiting ones. I couldn't find a way to dynamically choose dt with the actual parameter for this evaluation, as the dynamic size allocation of the array would clash with JAX. Not familiar with Greschgorin, would you mind explaining?